src/player.rs

```diff
@@ -0,0 +1,47 @@
+//! This module contains a barebones loader for track files.
+use crate::track::Track;
+use std::collections::HashMap;
+use std::fs::File;
+use std::path::Path;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Failed to open file for reading track data")]
+    OpenTrackFile(#[source] std::io::Error),
+    #[error("Failed to deserialize track data")]
+    DeserializeTracks(#[source] bincode::Error),
+}
+
+/// A loader for track binary files dumped by [`Client::save_tracks`](crate::Client::save_tracks).
+///
+/// # Usage
+///
+/// After constructing, call [`Player::get_track`] to get tracks.
+/// Then call [`Track::get_value`] to get saved values at any given point in time.
+pub struct Player {
+    tracks: HashMap<String, Track>,
+}
+
+impl Player {
+    /// Load track data from file for playback.
+    pub fn new(path: impl AsRef<Path>) -> Result<Self, Error> {
+        // Load from file
+        let file = File::open(path).map_err(Error::OpenTrackFile)?;
+        let tracks_vec: Vec<Track> =
+            bincode::deserialize_from(file).map_err(Error::DeserializeTracks)?;
```

Comment:
Do we want to decide for the user what format they want to serialize in, and if we do, do we want to document this somewhere?

Reply:
Another solution I can think of woud be to remove bincode entirely and make `save_tracks` and `RocketPlayer::new` just return and take a `Vec<Track>` (`Track` still implementing `Serialize` and `Deserialize` of course). Then the demo could decide how to store the tracks. Or maybe offer both options, `RocketClient::get_tracks` and `RocketPlayer::from_tracks` but also feature-gated `RocketClient::save_tracks` (maybe call the feature bincode). Also, serde itself could be a toggleable feature but does anyone ever __not__ wanna be able to save and load? Then there is also the possibility of taking Rocket editor XML-ish files, but I don't know if other editors support that format

Reply:
I like the idea of implementing `Serialize` and `Deserialize` of course. I do prefer not having bincode as a dependency at this stage, I'd rather have a `to_tracks` or something to turn a `RocketClient` into a Serializable thing, and have a `from_tracks` (or new) on `RocketPlayer`.

Reply:
If Tracks (or RocketData or something) was a type with just `Vec<Track>` inside, we could `impl From<RocketData> for Player` which I think is kinda neat (even though `from_tracks(Vec<Track>)` would work fine too). I'm just not sure if it's a good idea to turn `RocketClient` __into__ anything because wouldn't that then make it necessary to reconstruct the Client if the user wishes to keep editing? I think just giving the ability for the client to dump clones of it's Vec<Track> which can then be serialized regardless if the Rocket session is kept or destroyed would suit my use case better. I can also expand the examples to show reading and writing files (moving bincode to dev-dependencies for it to work)

Reply:
I think just cloning the Tracks is a more straightforward solution for now. As for bincode, it would be excellent to move that to dev deps and update the examples.

Reply:
Hmm or, what if the Client can be straight up turned into a Player and the Player can be (de)serialized?

Reply:
No, that would require the Client to have access to Player's internals or the Player to have a public contructor anyway. I think we'll roll with just save_tracks giving clones of Vec<Track> and Player::new taking Vec<Track>s. Thanks for the feedback again